numpy/core/src/multiarray/ctors.c

```diff
@@ -550,10 +553,7 @@ setArrayFromSequence(PyArrayObject *a, PyObject *s,
         PyObject *o;
```

Comment:
Tempting to do some cleanup at the same time
```suggestion
        PyObject *o = PySequence_Fast_GET_ITEM(seq, 0);;
```